test/soak_test/map_soak_test.js

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { Client, Predicates } = require('../../');
+const IdentifiedEntryProcessor = require('./IdentifiedEntryProcessor');
+const identifiedFactory = require('./IdentifiedFactory');
+
+const NS_IN_MS = 1e6;
+const MS_IN_SEC = 1e6;
+
+/**
+ * Test variables
+ */
+
+const MAX_ALLOWED_ACTIVE_REQUESTS = 32;
+const MAX_ALLOWED_CONSECUTIVE_REQUESTS = MAX_ALLOWED_ACTIVE_REQUESTS;
+let stopTest = false;
+let runningOperations = 0;
+let startTime;
+let endTime;
+let client;
+let totalOps = 0;
+let nz = 0;
+
+/**
+ * Helper functions
+ */
+
+function fancyDuration(millisec) {
+    const times = Math.floor(millisec / 1000);
+
+    const secs = Math.floor(times % 60);
+    const minutes = Math.floor((times % 3600) / 60);
+    const hours = Math.floor(times / 3600);
+
+    return hours + ':' + minutes + ':' + secs;
+}
+
+function hrtimeToNanoSec(t) {
+    return t[0] * NS_IN_MS + t[1];
```

Comment:
If I am not mistaken, `t[0]` is in seconds, but we are multiplying it with `NS_IN_MS`, I guess it should be `NS_IN_MS * 1e3`

Reply:
Did check this part of the ported code. Let me fix it.

Reply:
Addressed in e3d645832e65ecdeb577acf21812e846c5b39f9a